pym/bob/scm/checkoutAssert.py

```diff
@@ -0,0 +1,50 @@
+# Bob build tool
+# Copyright (C) 2018 Ralf Hubert
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import hashlib
+import schema
+import struct
+from textwrap import dedent
+
+class CheckoutAssert:
+    SCHEMA = schema.Schema({
+        'file' : str,
+        'digestSHA1' : str,
+        schema.Optional('start') : int,
+        schema.Optional('end') : int,
+    })
+
+    def __init__(self, spec):
+        self.__file = spec['file']
+        self.__digestSHA1 = spec['digestSHA1']
+        self.__start = spec.get('start', None)
+        self.__end = spec.get('end', None)
+
+    def asScript(self):
+        if self.__start is not None and self.__end is not None:
```

Comment:
What about only giving `start` *or* `end`? I think they should default to `1` resp. `last line`...